### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by SoorajModi**
**Why**

- Wanted to add details and improve wording on our docs

---

**What**

- Add `CI.md` to explain how to add CI pipelines
  - I feel like participants are going to encounter permission issues here, but im thinking this will be taken on by more experienced students and we can handle issues on a case by case basis
- Modify pre-requisites
- Adjusted some rules
- Tweaked categories
- Added a bit to the step by step guide

---

**Verification**

- N/A

---

**Checklist:**

- [x] This contribution abides by all rules listed in `docs/Rules.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ During the workshop, students will fork a public repository, add their contribut
 - GitHub account
     - If you are a student, you can sign up for the [GitHub Student Developer Pack](https://education.github.com/pack) to get free access to developer tools and resources.
 - Git installed on your computer
-- Basic programming skills
+    - See [Installing Git on Mac](https://github.com/SoorajModi/ContributeAnythingGryphHacks23/blob/main/docs/InstallGitOnMacOS.md) or [Installing Git on Windows](https://github.com/SoorajModi/ContributeAnythingGryphHacks23/blob/main/docs/InstallGitOnWindows.md)
 - Basic understanding of git recommended but not required
-- Read all rules for this workshop
+    - See [Git Basics](https://github.com/SoorajModi/ContributeAnythingGryphHacks23/blob/main/docs/GitBasics.md), or tutorials like [GitHowTo](https://githowto.com/) and [Learn Git Branching](https://learngitbranching.js.org/).
+- Read all rules for this workshop listed [here](https://github.com/SoorajModi/ContributeAnythingGryphHacks23/blob/main/docs/Rules.md)
```
Comment (MarshallAsch) [inline README.md]: I think we can use relative paths for these links so they will also work on forks

**Review by MarshallAsch — COMMENTED** [review]
Inital comment, I'll look at it again tomorrow

File: README.md (hunk 1 above)
Comment (SoorajModi) [reply README.md]: What would relative paths look like?

Comment (MarshallAsch) [reply README.md]: Links in github markdown do not need to be absolute and you can reference relative paths to files as shown here in thr blog post about it
https://github.blog/2013-01-31-relative-links-in-markup-files/

Comment (SoorajModi) [reply README.md]: Neat, updated to relative links!

**Review by MarshallAsch — APPROVED** [review]

**Pull request merged**